ci-automation/tapfile_helper_lib.sh

```diff
@@ -0,0 +1,288 @@
+#!/bin/bash
+#
+# Copyright (c) 2021 The Flatcar Maintainers.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+# Helper script for extracting information from TAP files and for merging multiple
+#  TAP files into one report.
+# The script uses a temporary SQLite DB for querzing and for result generation.
+#
+# Brief usage overview (scroll down for parameters etc.):
+#   tap_ingest_tapfile - add test results from tap file to the DB 
+#   tap_list_vendors   - list all vendors TAP files have been ingested for
+#   tap_failed_tests_for_vendor - list all tests that never succeded even once, per vendor
+#   tap_generate_report - generate a merged test report
+
+
+TAPFILE_HELPER_DBNAME="results.sqlite3"
```

Comment:
Is this expected to be shared across runs - e.g. if you can't run all vendor tests at the same time? If so then it should be a parameter passed by the caller.

Reply:
No, it's a transient / temporary throw-away database I use for structuring information from a bunch of TAP files into a query-able format.

Reply:
The TAP files of individual tests should be archived for a build, but not the sqlite file. The same tapfile helper can be used later (with a separate throw-away DB file) to merge all TAP files of all tests.